activerecord/lib/active_record/associations/preloader/through_association.rb

```diff
@@ -89,6 +90,10 @@ def through_scope
 
           scope
         end
+
+        def target_records_from_association(association)
+          association.loaded? ? association.target : association.reader
```

Comment:
@yuroyoro why do you check against `loaded?` here? It seems that association is always loaded in this case and all tests remain green when I delete it. Can you add a test that fails without it?